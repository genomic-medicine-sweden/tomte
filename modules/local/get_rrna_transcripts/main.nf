process GET_RRNA_TRANSCRIPTS {
    tag 'get_rrna_bed'
    label 'process_low'

    conda "bioconda::pirate=1.0.4 bioconda::perl-bioperl=1.7.2"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/pirate:1.0.4--hdfd78af_2' :
        'biocontainers/pirate:1.0.4--hdfd78af_2' }"

    input:
    tuple val(meta), path(gtf)

    output:
    path('rrna.gtf')    , emit: rrnagtf
    path('rrna.bed')    , emit: bed
    path "versions.yml" , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    """
    $baseDir/bin/get_rrna_transcripts.py --gtf ${gtf} --output rrna.gtf

    $baseDir/bin/gtf2bed rrna.gtf > rrna.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        get_rrna_transcripts: v1.0
    END_VERSIONS
    """

    stub:
    """
    touch rrna.gtf
    touch rrna.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        get_rrna_transcripts: v1.0
    END_VERSIONS
    """
}
