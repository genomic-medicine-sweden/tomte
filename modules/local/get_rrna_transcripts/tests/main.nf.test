nextflow_process {

    name "Test Process GET_RRNA_TRANSCRIPTS"
    script "../main.nf"
    process "GET_RRNA_TRANSCRIPTS"
    tag "modules_local"
    tag "modules"
    tag "get_rrna_transcripts"

    test("Should run without failures") {

        when {
            params {
                modules_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
            }
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.gtf', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
