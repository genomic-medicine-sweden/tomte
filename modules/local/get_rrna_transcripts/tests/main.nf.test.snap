{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "rrna.gtf:md5,4badbbf7f1857664fb54616234c7fd86"
                ],
                "1": [
                    "rrna.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "versions.yml:md5,2cf86a5a74d87d2bf026f99e89e624cf"
                ],
                "bed": [
                    "rrna.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "rrnagtf": [
                    "rrna.gtf:md5,4badbbf7f1857664fb54616234c7fd86"
                ],
                "versions": [
                    "versions.yml:md5,2cf86a5a74d87d2bf026f99e89e624cf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-15T16:28:48.240325"
    }
}